src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -501,28 +393,112 @@ private static MavenArtifactType GetMavenArtifactType(string fileName)
                 // Split the version content into its individual parts.
                 // [0] - The version
                 // [1] - The date it was published
-                // [2] - The time it waas published
-                // [3] - The download count
+                // [2] - The time it was published
+                // [3] - The file size in bytes
                 string[] versionParts = versionContent.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
                 if (DateTime.TryParse($"{versionParts[1]} {versionParts[2]}", out DateTime publishDateTime))
                 {
                     return publishDateTime;
                 }
             }
-            else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+
+            // If the directory index approach does not work, try to get the "Last-Modified" header from the .pom file.
+            baseUrl = $"{feedUrl.EnsureTrailingSlash()}{packageNamespace}/{packageName}/{packageVersion}/{packageName}-{packageVersion}.pom";
+            HttpResponseMessage response = await httpClient.GetAsync(baseUrl);
+
+            if (response.Content.Headers.TryGetValues("Last-Modified", out var values))
+            {
+                var lastModified = DateTime.Parse(values.First());
+                return lastModified;
+            }
+
+            return null;
+        }
+
+        private async IAsyncEnumerable<string> GetArtifactDownloadUris_DirectoryIndexStrategyAsync(string baseUrl, bool useCache = true)
+        {
+            HttpClient httpClient = CreateHttpClient();
+            var fileNameList = new List<string>();
+
+            try
+            {
+                string? html = await GetHttpStringCache(httpClient, baseUrl, useCache);
+                if (string.IsNullOrEmpty(html))
+                {
+                    throw new InvalidOperationException();
+                }
+
+                HtmlParser parser = new();
+                AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(html);
+
+                foreach (string fileName in document.QuerySelectorAll("a").Select(link => link.GetAttribute("href").ToString()))
+                {
+                    if (fileName == "../") continue;
+                    fileNameList.Add(baseUrl + fileName);
+                }
+            }
+            catch {}
```

Comment:
How would we debug an unexpected exception from `GetHttpStringCache`?

Reply:
Same note applies to the other strategies as well.

Reply:
I added some additional exception handling and logging to `GetArtifactDownloadUris_FileProbeStrategyAsync` and `GetArtifactDownloadUris_ArtifactMetadataStrategyAsync`.